src/Tizen.Multimedia/AudioManager/AudioDevice.cs

```diff
@@ -104,6 +107,185 @@ public bool IsRunning
+        {
+            IntPtr rates;
+            uint numOfElems;
+
+            Interop.AudioDevice.GetSupportedSampleRates(_id, out rates, out numOfElems).
+                ThrowIfError("Failed to get supported sample formats");
+
+            return RetrieveRates();
+
+            IEnumerable<uint> RetrieveRates()
+            {
+                int[] ratesRes = new int[numOfElems];
+
+                Marshal.Copy(rates, ratesRes, 0, (int)numOfElems);
+                Interop.Libc.Free(rates);
+
+                IEnumerable<uint> res = ratesRes.OfType<uint>();
+                foreach (uint r in res)
+                {
+                    Log.Debug(Tag, $"supported sample rate:{r}");
+                }
+
+                return res;
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the device's sample rate.
+        /// </summary>
+        /// <remarks>
+        /// This device should be <see cref="AudioDeviceType.UsbAudio"/> type and <see cref="AudioDeviceIoDirection.Output"/> direction.
+        /// </remarks>
+        /// <exception cref="ArgumentException">This device is not valid or is disconnected.</exception>
+        /// <exception cref="InvalidOperationException">This device is not <see cref="AudioDeviceType.UsbAudio"/> type or is not <see cref="AudioDeviceIoDirection.Output"/> direction.</exception>
+        /// <since_tizen> 5 </since_tizen>
+        public uint SampleRate
+        {
+            get
+            {
+                Interop.AudioDevice.GetSampleRate(_id, out uint rate).
+                    ThrowIfError("Failed to get sample rate of the device");
+
+                return rate;
+            }
+            set
+            {
+                Interop.AudioDevice.SetSampleRate(_id, value).
+                    ThrowIfError("Failed to set sample rate of the device");
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the device's 'avoid resampling' property.
+        /// </summary>
+        /// <remarks>
+        /// This device should be <see cref="AudioDeviceType.UsbAudio"/> type and <see cref="AudioDeviceIoDirection.Output"/> direction.
+        /// This property is not enabled as default. With this enabled, this device will use the first stream's original sample format
+        /// and rate without resampling if supported.
+        /// </remarks>
+        /// <exception cref="ArgumentException">This device is not valid or is disconnected.</exception>
+        /// <exception cref="InvalidOperationException">This device is not <see cref="AudioDeviceType.UsbAudio"/> type or is not <see cref="AudioDeviceIoDirection.Output"/> direction.</exception>
+        /// <since_tizen> 5 </since_tizen>
+        public bool AvoidResampling
+        {
+            get
+            {
+                Interop.AudioDevice.GetAvoidResampling(_id, out bool enabled).
+                    ThrowIfError("Failed to get avoid-resampling property of the device");
```

Comment:
X AVOID throwing exceptions from property getters. 
see https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/property

Reply:
I'll revise it with set/get methods instead of it.